src/wordvec.c

```diff
@@ -0,0 +1,50 @@
+#define _GNU_SOURCE
+#include "wordvec.h"
+
+struct wordvec *wordvec_new(void)
+{
+	struct wordvec *wv = malloc(sizeof(*wv));
+	if (!wv)
+		return NULL;
+
+	wv->len = 0;
+	wv->cap = WORDVEC_DEFAULT_CAP;
+
+	wv->vec = calloc(wv->cap, sizeof(*wv->vec));
+	if (!wv->vec) {
```

Comment:
Merge this allocation failure with the other one? According to ``man 3 free``:
> If ptr is NULL, no operation is performed

Reply:
`wv->vec` causes a dereference, we need to first check `wv`

Reply:
I was talking something like:
``` c
... wv = malloc(....);
... wv_vec = calloc(...);
if (!wv || !wv_vec) {
    free(wv);
    free(wv_vec);
}
```

But I'm ok with your implementation